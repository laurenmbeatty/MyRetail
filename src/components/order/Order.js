import React from "react";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import * as Styled from "./styles";
class Order extends React.Component {
  state = {
    quantity: 1
  };

  decrement = () => {
    this.state.quantity === 0
      ? this.setState({
          quantity: 0
        })
      : this.setState({
          quantity: this.state.quantity - 1
        });
  };

  increment = () => {
    this.setState({
      quantity: this.state.quantity + 1
    });
  };

  render() {
    const { quantity } = this.state;
    const { available } = this.props;
    return (
      <Styled.Order>
        <Styled.OrderForm>
          <Styled.Quantity>
            <label className="quantity-label" htmlFor="quantity">
              quantity:
            </label>
            <div className="counter-container">
              <FontAwesomeIcon
                icon="minus-circle"
                onClick={this.decrement}
                className="decrement-button"
              />
              <span className="quantity">{quantity}</span>
              <FontAwesomeIcon
                icon="plus-circle"
                onClick={this.increment}
                className="increment-button"
              />
            </div>
          </Styled.Quantity>
          <Styled.OrderButtonContainer>
            {available === "0" || available === "1" ? (
              <button className="pick-up">PICK UP IN STORE</button>
            ) : null}
            {available === "0" || available === "2" ? (
              <button className="add-to-cart">ADD TO CART</button>
            ) : null}
          </Styled.OrderButtonContainer>
          {available === "0" || available === "1" ? (
            <a
              href="https://www.target.com/s?searchTerm=ninja+professional+1000+watt+blender"
              rel="noopener noreferrer"
              target="_blank"
              className="find-in-store"
            >
              find in a store
            </a>
          ) : null}
        </Styled.OrderForm>
        <Styled.Returns>
          <p className="returns-label">returns</p>
          <hr />
          <p className="returns-description">
            This item must be returned within 30 days of the ship date. See
            return policy for details. Prices, promotions, styles and
            availability may vary by store and online.
          </p>
        </Styled.Returns>
        <Styled.SocialButtonContainer>
          <button className="social-button">ADD TO REGISTRY</button>
          <button className="social-button">ADD TO LIST</button>
          <button className="social-button">SHARE</button>
        </Styled.SocialButtonContainer>
      </Styled.Order>
    );
  }
}
export default Order;
