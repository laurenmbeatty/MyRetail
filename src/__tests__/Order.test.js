import React from "react";
import { create } from "react-test-renderer";
import { render, cleanup } from "react-testing-library";
import "jest-styled-components";
import Order from "../components/order/Order";

afterEach(cleanup);

test("snapshot", () => {
  const c = create(<Order />);
  expect(c.toJSON()).toMatchSnapshot();
});

test("increments when increment function is called", () => {
  const c = create(<Order increment={() => {}} decrement={() => {}} />);
  const instance = c.getInstance();

  expect(instance.state.quantity).toBe(1);
  instance.increment();
  expect(instance.state.quantity).toBe(2);
  instance.decrement();
  instance.decrement();
  instance.decrement();
  expect(instance.state.quantity).toBe(0);
});

test("Quantity does not go below zero when decrement is called", () => {
  const c = create(<Order increment={() => {}} decrement={() => {}} />);
  const instance = c.getInstance();

  expect(instance.state.quantity).toBe(1);
  instance.decrement();
  instance.decrement();
  instance.decrement();
  expect(instance.state.quantity).toBe(0);
});

test("renders the pick up in store button and add to cart button if available equals 0", () => {
  const { getByText } = render(<Order available="0" />);

  const pickUpButton = getByText(/PICK UP IN STORE/);
  const addToCartButton = getByText(/ADD TO CART/);

  expect(pickUpButton).toBeDefined();
  expect(addToCartButton).toBeDefined();
});
