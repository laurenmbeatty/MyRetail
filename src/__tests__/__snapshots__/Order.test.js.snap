// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
.c0 {
  color: #ccc;
  margin-top: 30px;
  width: 100%;
}

.c0 .find-in-store {
  color: #000;
  font-family: Arial,Helvetica,sans-serif;
  font-size: 12px;
  font-weight: bold;
  line-height: 1.17;
  -webkit-text-decoration: none;
  text-decoration: none;
  display: inline-block;
  width: 50%;
  text-align: center;
}

.c1 {
  width: 100%;
}

.c2 {
  border: 1px solid #d3d2d2;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 5px 8px 5px 11px;
  width: 45%;
}

.c2 .quantity-label {
  color: #333;
  font-family: Arial,Helvetica,sans-serif;
  font-size: 14px;
}

.c2 .counter-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c2 .quantity {
  font-family: Arial,Helvetica,sans-serif;
  font-size: 18px;
  color: #000;
  padding-right: 13px;
}

.c2 svg {
  font-size: 30px;
  cursor: pointer;
}

.c2 .decrement-button {
  padding-right: 13px;
}

.c3 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 31px;
}

.c3 button {
  padding-top: 10px;
  padding-bottom: 10px;
  width: 49%;
  background-image: none;
  border-radius: 3px;
  box-shadow: 0px 1px 2px 0 rgba(0,0,0,0.5);
  font-family: Helvetica,sans-serif;
  font-size: 18px;
  line-height: 1.03;
  color: #fff;
}

.c3 .pick-up-container {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c3 .pick-up {
  background-color: #000;
  border-color: #000;
}

.c3 .add-to-cart {
  background-color: #cc0000;
  border-color: #cc0000;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-top: 10px;
}

.c4 .returns-label {
  font-family: Arial,Helvetica,sans-serif;
  font-size: 20px;
  line-height: 0.9;
  color: #666;
  padding-right: 15px;
}

.c4 .returns-description {
  font-family: Arial,Helvetica,sans-serif;
  font-size: 10px;
  line-height: 1.4;
  color: #666;
  padding-left: 14px;
}

.c4 hr {
  -webkit-transform: rotate(180deg);
  -ms-transform: rotate(180deg);
  transform: rotate(180deg);
  height: 20px;
  color: #ccc;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-top: 13px;
}

.c5 .social-button {
  padding-top: 4px;
  padding-bottom: 5px;
  font-family: Arial,Helvetica,sans-serif;
  font-size: 12px;
  background-color: #f3f3f3;
  color: #333333;
  border: none;
  width: 32%;
}

<section
  className="c0"
>
  <form
    className="c1"
  >
    <div
      className="c2"
    >
      <label
        className="quantity-label"
        htmlFor="quantity"
      >
        quantity:
      </label>
      <div
        className="counter-container"
      >
        <span
          className="quantity"
        >
          1
        </span>
      </div>
    </div>
    <div
      className="c3"
    />
  </form>
  <div
    className="c4"
  >
    <p
      className="returns-label"
    >
      returns
    </p>
    <hr />
    <p
      className="returns-description"
    >
      This item must be returned within 30 days of the ship date. See return policy for details. Prices, promotions, styles and availability may vary by store and online.
    </p>
  </div>
  <div
    className="c5"
  >
    <button
      className="social-button"
    >
      ADD TO REGISTRY
    </button>
    <button
      className="social-button"
    >
      ADD TO LIST
    </button>
    <button
      className="social-button"
    >
      SHARE
    </button>
  </div>
</section>
`;
